code/modules/food_and_drinks/restaurant/_venue.dm

```diff
@@ -184,20 +187,27 @@
 
 /obj/machinery/restaurant_portal/Initialize(mapload)
 	. = ..()
-	if(linked_venue)
-		linked_venue = SSrestaurant.all_venues[linked_venue]
-		linked_venue.restaurant_portal = src
 	register_context()
+	if (!linked_venue)
+		return
```

Comment:
when would this be valid?

Reply:
mapped-in types have linked_venue set to a typepath, player-built machines don't so we abort it here.